data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/ScanObjectWorker.java

```diff
@@ -99,7 +100,18 @@ public ScanObjectWorker(final S3Client s3Client,
     @Override
     public void run() {
         while (!shouldStopProcessing) {
-            startProcessingObject(STANDARD_BACKOFF_MILLIS);
+
+            try {
+                startProcessingObject(STANDARD_BACKOFF_MILLIS);
+            } catch (final Exception e) {
+                LOG.error("Received an exception while processing s3 objects, backing off and retrying", e);
+                try {
+                    Thread.sleep(RETRY_BACKOFF_ON_EXCEPTION_MILLIS);
+                } catch (InterruptedException ex) {
+                    ex.printStackTrace();
```

Comment:
Let's not call `ex.printStackTrace()`. Please replace with:

```
LOG.error("Thread interrupted while backing off.", ex);
```

Or some other text of your choosing.